### PR TITLE
[osgearth] Update to 3.5

### LINKED ISSUE #34689: [<osgEarth>] update to <3.5>
### Library name

osgEarth

### New version number

3.5

### Other information that may be useful (release notes, etc...)

https://github.com/gwaldron/osgearth
https://github.com/gwaldron/osgearth/releases
https://github.com/gwaldron/osgearth/archive/refs/tags/osgearth-3.5.tar.gz

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fixes #34689.
All features passed with following triplets:
```
x64-windows
x64-windows-static
```
Usage test passed on `x64-windows`.

### CODE REVIEW

File: ports/osgearth/vcpkg.json [hunk 1]
```diff
@@ -25,6 +24,7 @@
     },
     "protobuf",
     "pthreads",
+    "spdlog",
```
Comment (dg0yt) [inline ports/osgearth/vcpkg.json:27]: Is this a hard dependency?

**Review by dg0yt — COMMENTED** [review]
Based on the new support for per-feature `platform` expressions, the `default-features` feature should be unrolled.

File: ports/osgearth/vcpkg.json (hunk 1 above)
Comment (LilyWangLL) [reply ports/osgearth/vcpkg.json]: No, the following message are from upstream, I removed it.
```
Notify: integrate spdlog as an optional component for logging
```

**Comment by JonLiu1993** [comment]
Pinging @LilyWangLL  for response. Is work still being done for this PR?

**Comment by HorrZzz** [comment]
https://github.com/gwaldron/osgearth/commit/7cb2f3dbb20eb7681e2550e3c9e1e404aea3a2d5

Has it been resolved yet?

**Comment by gwaldron** [comment]
Not yet - working on a patch release - please stand by.

**Comment by JonLiu1993** [comment]
Duplicate of https://github.com/microsoft/vcpkg/pull/41322

**Pull request closed without merging**